requirements/base.txt

```diff
@@ -7,3 +7,4 @@ optree >=0.12.1
 opt_einsum >= 3.3.0
 mpmath <1.4.0  # todo: teporarl pin for `NameError: name '_C' is not defined`
 dill >=0.3.8 # Support for 3.12
+black
```

Comment:
We don't have to add `black` as a dependency; I'll comment below about how to deal with it